diff --git a/lua/zincoxide.lua b/lua/zincoxide.lua
index 88cf435..5d6cbe7 100644
--- a/lua/zincoxide.lua
+++ b/lua/zincoxide.lua
@@ -262,6 +262,9 @@ function M.cd(opts, behaviour)
   elseif behaviour == "global" then
     vim.cmd("cd " .. target)
   end
+
+  local current_dir = vim.fn.getcwd()
+  vim.notify("Current directory: " .. current_dir, vim.log.levels.INFO, { title = "Directory changed" })
 end
 
 return M
